core/templates/dev/head/pages/exploration-editor-page/services/exploration-rights-data.service.ts

```diff
@@ -0,0 +1,46 @@
+// Copyright 2019 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview A data service that stores data
+ * about the rights for this exploration.
+ * This is different from the exploration rights service
+ * since the data used here is required globally and may
+ * create a circular dependency with exploration data service
+ * which is used in exploration rights service
+ */
+require('services/ContextService.ts');
+
+var oppia = require('AppInit.ts').module;
+
+oppia.factory('ExplorationRightsDataService', [
+  '$http', '$q', 'ContextService',
+  function($http, $q, ContextService) {
+    var rights = null;
+    return {
+      getRightsAsync: function() {
+        if (rights) {
+          return $q.resolve(rights);
+        }
+        return $http.get(
+          '/createhandler/rights/' + ContextService.getExplorationId()).then(
+          function(response) {
+            rights = response.data;
+            return $q.resolve(rights);
+          }
```

Comment:
Can we cache the promise itself instead of the data? (As was done with csrf)

Reply:
Done